extensions/libarchive/CompressContent.cpp

```diff
@@ -35,10 +35,32 @@ namespace nifi {
 namespace minifi {
 namespace processors {
 
-core::Property CompressContent::CompressLevel("Compression Level", "The compression level to use; this is valid only when using GZIP compression.", "1");
-core::Property CompressContent::CompressMode("Mode", "Indicates whether the processor should compress content or decompress content.", MODE_COMPRESS);
-core::Property CompressContent::CompressFormat("Compression Format", "The compression format to use.", COMPRESSION_FORMAT_ATTRIBUTE);
-core::Property CompressContent::UpdateFileName("Update Filename", "Determines if filename extension need to be updated", "false");
+core::Property CompressContent::CompressLevel(
+    core::PropertyBuilder::createProperty("Compression Level")->withDescription("The compression level to use; this is valid only when using GZIP compression.")
+        ->isRequired(false)->withDefaultValue<int>(1)->build());
+core::Property CompressContent::CompressMode(
+    core::PropertyBuilder::createProperty("Mode")->withDescription("Indicates whether the processor should compress content or decompress content.")
+        ->isRequired(false)->withAllowableValues<std::string>({MODE_COMPRESS, MODE_DECOMPRESS})->withDefaultValue(MODE_COMPRESS)->build());
+core::Property CompressContent::CompressFormat(
+    core::PropertyBuilder::createProperty("Compression Format")->withDescription("The compression format to use.")
+        ->isRequired(false)
+        ->withAllowableValues<std::string>({
+          COMPRESSION_FORMAT_ATTRIBUTE,
+          COMPRESSION_FORMAT_GZIP,
+          COMPRESSION_FORMAT_BZIP2,
+          COMPRESSION_FORMAT_XZ_LZMA2,
```

Comment:
Could you elaborate? Any YAML config that didn't use values from this enum was erroneous and resulted in an error on onTrigger. How this change would break those configurations further?

Reply:
In that case, I was wrong.